### PR TITLE
Implement views for debit card payment via GOV.UK pay

### PR DESCRIPTION
**Pull request by ibrechin**
Handles interaction with the MTP API and the GOV.UK Pay API
in order to create a transaction, forward the user to the
payment site, and receive the user after payment completion,
updating the transaction in the backend.

### CODE REVIEW

**Comment by ibrechin** [comment]
retest this please

File: mtp_send_money/apps/send_money/views.py [hunk 1]
```diff
@@ -92,19 +99,74 @@ def bank_transfer_view(request):
...
+        }
+
+        govuk_response = requests.post(
+            govuk_url('/payments'), headers=govuk_headers(), data=new_payment
+        )
+
+        if govuk_response.status_code == 201:
+            return redirect(get_link_by_rel(govuk_response.json(), 'next_url')['href'])
+        else:
+            logger.error(
```
Comment (ushkarev) [inline mtp_send_money/apps/send_money/views.py]: we'd probably want to present the user with some error message here. e.g. "service temporarily unavailable", but maybe that's a later story

File: mtp_send_money/apps/send_money/views.py [hunk 2]
```diff
@@ -92,19 +99,74 @@ def bank_transfer_view(request):
...
-    # - check payment status
-    # - present errors or redirect to start form?
-    # - present confirmation message?
-    # - present payment pending message? how would user check back?
+    ref = request.GET.get('paymentReference')
+    if ref is None:
+        return redirect(reverse('send_money:send_money'))
+    context = {'success': False}
+
+    govuk_response = requests.get(
```
Comment (ushkarev) [inline mtp_send_money/apps/send_money/views.py]: possibly some try-except would be good here. i wonder if we can make some generic middleware or view decorator to catch gov.uk pay exceptions and present some sort of "service temporarily not available" msg

File: mtp_send_money/apps/send_money/views.py (hunk 1 above)
Comment (ibrechin) [inline mtp_send_money/apps/send_money/views.py]: it falls through to rendering the 'failure.html' page, which will have some message of that form on it

File: mtp_send_money/settings/base.py [hunk 3]
```diff
@@ -23,6 +23,7 @@
 SECRET_KEY = 'CHANGE_ME'
 ALLOWED_HOSTS = []
 
+SITE_URL = os.environ.get('SITE_URL', 'localhost')
```
Comment (ushkarev) [inline mtp_send_money/settings/base.py]: is this in the deploy repo? i like the idea, we should add it to all client apps

Comment (ibrechin) [inline mtp_send_money/settings/base.py]: Will be adding to deploy repo.

Probably won't go through adding it to all the client apps until we actually need it, unless you have a use case in mind.

Comment (ushkarev) [inline mtp_send_money/settings/base.py]: i just like consistency :)

**Pull request merged**